### DISCUSSION TITLE
[ZK Stack] - Changing Data Availability Layer

### DISCUSSION THREAD

**Discussion by 6789tc** [zkSync-Community-Hub/zksync-developers#486] [ZK Stack] [answered]
### Module Affected

- [X] System contracts (era-system-contracts)
- [X] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [X] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

We are currently using ZK Stack to enable builders to spin up their ZK Rollups on Bitcoin via an intermediate EVM-compatible Layer 1 called BVM (https://bvm.network/).

In the original ZK Stack flow, a batch of transactions is directly stored on Layer 1 in blobs. However, in BVM, we would like to store the batch in alternative Data Availability (DA) storages such as Celestia, Polygon, etc.

We've made the following changes to the ZK Stack's source code to achieve this:

- In the configuration file, we changed the setting from "Blob" to "Calldata": `ETH_SENDER_SENDER_PUBDATA_SENDING_MODE=Calldata`.

- At the commit batch step, we replaced `pubdata` with `hash(pubdata)` and submitted the `pubdata` to an alternative DA.

- We then wait for the DA to confirm the `pubdata`'s inclusion before sending a commit batch transaction to Layer 1.

- We also modified the `verify` function of the Layer 1 contract to ignore the `pubdata`, essentially skipping the check: `logOutput.pubdataHash == keccak256(_newBatch.pubdataCommitments[1:_newBatch.pubdataCommitments.length - 32])`.

Our question is whether there are any security concerns with the changes we made, as described above, and if so, please advise us on how to address them.

Thanks in advance.

### Expected Behavior

There is a way to change Data Availability layer for storing batches of transactions instead of storing them directly on Layer 1

### Current Behavior

The batches of transactions are directly stored in Layer 1

### Repository Link (if applicable)

https://github.com/TrustlessComputer/zk-rollups

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by bxpana** [comment]
Thanks for the post, we'll get back asap

**Reply by bxpana** [reply, marked as answer]
We are currently working on having other DA options available for ZK Stack, but it's not quite available yet. 

Regarding any security concerns, it would be best to get an audit to make sure that there would be no issues with the changes that you made.
